### DISCUSSION TITLE
Server instance

### DISCUSSION THREAD

**Discussion by gkielwasser** [directus/directus#9603] [Archive]
Hi,

I did the migration to Directus 9.0.0. My Websocket registration is not working anymore.

I used to register Socket.io this way previously using hooks:
```
'server.start.before': async function ({ server }) {
	const io = new SocketIoServer(server);
},
```

First thing I have noticed is the 'before' seems not anymore supported as I cannot see anymore in the 'filter' section.

Secondly, the 'action' is well called but there is not anymore the server parameter.

So my question is, how can I retrieve the Directus server instance with 9.0.0 to start properly Socket.io ?

Thanks

**Comment by joselcvarela** [comment]
Hello @gkielwasser 
Custom hooks have changed as mentioned here:
https://github.com/directus/directus/blob/main/changelog.md#%EF%B8%8F-breaking-changes

You can find more about new custom hooks API on documentation: 
https://docs.directus.io/extensions/hooks/#custom-api-hooks

In your case I believe you would need something like `init('app.before')`

**Reply by gkielwasser** [reply]
Hi @joselcvarela,
I just tried this hook but I don't get it how I could get the http server instance from the "app" argument I receive. I just receive an express request handler function.
Thanks for help

**Reply by joselcvarela** [reply]
Indeed, it seems we are not exposing server anymore.

Do you like to open a PR?
I believe you just need to pass server on 2nd argument: https://github.com/directus/directus/blob/a16703fc81c4b6e37855de4879140ac0b2598239/api/src/server.ts#L142-L150

**Reply by gkielwasser** [reply]
Thanks,

here is a [PR](https://github.com/directus/directus/pull/9616) about that
